src/com/rcv/RcvGui.java

```diff
@@ -0,0 +1,71 @@
+package com.rcv;
+
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.awt.BorderLayout;
+import javax.swing.*;
+import javax.swing.filechooser.FileNameExtensionFilter;
+import javax.swing.filechooser.FileSystemView;
+
+class RcvGui {
+  private static ElectionConfig config;
+  private static final JLabel labelStatus;
+  private static final JFileChooser fc;
+  private static final JFrame frame;
+  private static final FileNameExtensionFilter filterJson;
+
+  static {
+    labelStatus = new JLabel("Welcome to the Universal RCV Tabulator!");
+    fc = new JFileChooser(FileSystemView.getFileSystemView().getHomeDirectory());
+    frame = new JFrame("Universal RCV Tabulator");
+    filterJson = new FileNameExtensionFilter("JSON file", "json");
+  }
+
+  void launch() {
+    frame.setDefaultCloseOperation(WindowConstants.EXIT_ON_CLOSE);
+
+    JButton buttonLoadConfig = new JButton("Load config");
+    buttonLoadConfig.addActionListener(new LoadConfigListener());
+
+    JButton buttonTabulate = new JButton("Tabulate");
+    buttonTabulate.addActionListener(new TabulateListener());
+
+    JPanel panelMain = new JPanel();
+    frame.setContentPane(panelMain);
+
+    frame.getContentPane().add(BorderLayout.NORTH, buttonLoadConfig);
+    frame.getContentPane().add(BorderLayout.CENTER, buttonTabulate);
+    frame.getContentPane().add(BorderLayout.SOUTH, labelStatus);
+
+    // TODO: Make below a % of window size if possible
+    frame.setSize(800, 800);
+    frame.setVisible(true);
+
+    fc.setDialogTitle("Select a config file");
+    fc.setAcceptAllFileFilterUsed(false);
+    fc.addChoosableFileFilter(filterJson);
+  }
+
+  class LoadConfigListener implements ActionListener {
+    public void actionPerformed(ActionEvent event) {
+      int returnValue = fc.showOpenDialog(null);
+      if (returnValue == JFileChooser.APPROVE_OPTION) {
+        String configPath = fc.getSelectedFile().getAbsolutePath();
+        config = Main.makeElectionConfig(configPath);
+        Logger.log("Tabulator is being used via the GUI.");
+        if (config == null) {
+          labelStatus.setText(String.format("ERROR: Unable to load config file: %s", configPath));
+        } else {
+          labelStatus.setText(String.format("Successfully loaded config file: %s", configPath));
+        }
+      }
+    }
+  }
+
+  class TabulateListener implements ActionListener {
+    public void actionPerformed(ActionEvent event) {
+      String response = Main.executeTabulation(config);
+      labelStatus.setText(response);
+    }
+  }
+}
```

Comment:
Looks good.  The EAC coding standards are a bit pedantic in that they want: 
a header comment block for every function including purpose, params and return
also a comment for EVERY variable (only exception is loop index variables)
and a comment for EVERY constant (except 1, -1, and 0)

Reply:
Good to know, thanks. What if I document a function like this:

`// returns: String indicating whether or not execution was successful`
`static String executeTabulation(ElectionConfig config) {`

Do I also need to apply the same comment inside the function when actually defining the response variable?

`// String indicating whether or not execution was successful`
`String response = "Tabulation successful!";`

And then again when I actually call the function?

`// String indicating whether or not execution was successful`
`String response = Main.executeTabulation(config);`